sphinx_needs/builder.py

```diff
@@ -158,3 +157,69 @@ def build_needumls_pumls(app: Sphinx, _exception: Exception) -> None:
         needs_builder.set_environment(env)
 
     needs_builder.finish()
+
+
+class NeedsIdBuilder(Builder):
+    name = "needs_id"
+    format = "json"
+    file_suffix = ".txt"
+    links_suffix = None
+    LIST_KEY_EXCLUSIONS_NEEDS = ["content_node"]
+
+    def write_doc(self, docname: str, doctree: nodes.document) -> None:
+        pass
+
+    def finish(self) -> None:
+        env = unwrap(self.env)
+        needs = env.needs_all_needs.values()
+        needs_per_id_build_path = self.app.config.needs_per_id_build_path
+        filter_string = self.app.config.needs_builder_filter
+        filtered_needs = filter_needs(self.app, needs, filter_string)
+        needs_id_json_dir = os.path.join(self.outdir, needs_per_id_build_path)
+        if not os.path.exists(needs_id_json_dir):
+            os.mkdir(needs_id_json_dir)
+        for need in filtered_needs:
+            needs_id_dict = {}
+            id = need["id"]
+            needs_id_dict[id] = {key: need[key] for key in need if key not in self.LIST_KEY_EXCLUSIONS_NEEDS}
+            try:
+                fname = os.path.join(needs_id_json_dir, f"{id}.json")
+                with open(fname, "w") as f:
+                    json.dump(needs_id_dict, f, indent=4)
+            except Exception as e:
+                log.error(f"Error during writing json file: {e}_{id}")
```

Comment:
`{e}_{id}`  makes no sense as error message, as `e` could be really long.
Put the id at the beginning of the whole string itself.

Something like:
`f"Needs-ID Builder {id} error: {e}"`